python/cm/management/commands/loadcluster.py

```diff
@@ -119,16 +120,35 @@ def create_provider(provider):
         same_name_provider = models.HostProvider.objects.get(name=provider['name'])
         if same_name_provider.prototype.bundle.hash != bundle_hash:
             raise IntegrityError('Name of provider already in use in another bundle')
+        create_needed_file(same_name_provider, provider['config'])
         return ex_id, same_name_provider
     except models.HostProvider.DoesNotExist:
         prototype = get_prototype(bundle_hash=bundle_hash, type='provider')
-        config = create_config(provider.pop('config'))
+        config = provider.pop('config')
         provider = models.HostProvider.objects.create(
-            prototype=prototype, config=config, **provider
+            prototype=prototype, config=create_config(config), **provider
         )
+        create_needed_file(provider, config)
         return ex_id, provider
 
 
+def create_needed_file(obj, config):
+    conf = config["current"]["config"]
+    proto = obj.prototype
+    for key, value in conf.items():
+        if isinstance(value, dict):
+            for subkey, subvalue in value.items():
+                pconf = models.PrototypeConfig.objects.get(
```

Comment:
It may be look like premature optimization, but anyway:
`if models.PrototypeConfig.objects.filter(...., type='file').exists():
    save_file_type(...)
`

Reply:
Done in next commit